apps/utils/locust_tests/high_volume_suite.py

```diff
@@ -60,24 +71,45 @@ def _(environment: Environment, **kwargs):
         data_type_name="hashed_mbis",
     )
 
+
 class TestLoadShape(UserInitAwareLoadShape):
     pass
 
+
 class TaskHolder(Protocol[TaskT]):
     tasks: List[TaskT]
 
+
+class HighVolumeTaskSet(TaskSet):
+    @property
+    def user(self) -> "HighVolumeUser":
+        # This forces the type of self.user for all derived TaskSets to narrow to HighVolumeUser,
+        # thus giving correct type hinting for all of HighVolumeUser's properties.
+        return self._high_volume_user
+
+    def __init__(self, parent: User) -> None:
+        if not isinstance(parent, HighVolumeUser):
+            raise ValueError(
+                f"User is {type(self.user).__name__}; expected {type(HighVolumeUser).__name__}"
+            )
+        self._high_volume_user: HighVolumeUser = parent
+        super().__init__(parent)
+
+
 EOB_TAG = "eob"
+
+
 @tag(EOB_TAG)
 @task
-class EobTaskSet(TaskSet):
+class EobTaskSet(HighVolumeTaskSet):
     @tag("eob_test_id_count_type_pde_v1", "v1")
     @task
     def eob_test_id_count_type_pde_v1(self):
         """Explanation of Benefit search by ID, type PDE, paginated"""
-        self.run_task_by_parameters(
+        self.user.run_task_by_parameters(
             base_path="/v1/fhir/ExplanationOfBenefit",
             params={
-                "patient":  self.bene_ids.pop(),
+                "patient": random.choice(self.user.bene_ids),
```

Comment:
[`random.choice(...)`](https://docs.python.org/3/library/functions.html?highlight=next#next) is used to pick a random element from the enumerable. This ensures the source data collection is not mutated, nor exhausted, and that data is picked at random; therefore, this is equivalent to using `random.shuffle()` without needing to be concerned with data exhaustion.

Reply:
If i'm understanding this right, I think we shouldn't be applying the new random choice pattern to the high volume suite. For the regression suite we don't care about reusing data, but the high volume suite does, we want to avoid calling anything twice to avoid cached hits influencing the performance result numbers.

Reply:
I'm not sure how big of an impact that would have on the High Volume Suite's characteristics, but you are right; using `random.choice()` invites the possibility that a single `HighVolumeUser` _could_ re-use data that had already been used. And, the High Volume Suite doesn't use the limited synthetic datasets, so thinking about it now I don't think we should be concerned about data exhaustion at all.

I've reverted this change to the High Volume Suite. It now `shuffle`s each `list` of data, and data is `pop`ped from each `list` when it's used, effectively making it impossible for any `HighVolumeUser` (individually) to reuse data. This does mean that a `HighVolumeUser` could run out of data, but solving that is out-of-scope of this PR anyways.